prow/pjutil/pjutil.go

```diff
@@ -239,6 +239,44 @@ func GetLatestProwJobs(pjs []prowapi.ProwJob, jobType prowapi.ProwJobType) map[s
 	return latestJobs
 }
 
+// ClonePJ clones a ProwJob using the definition from the JobConfig.
+func ClonePJ(pj *prowapi.ProwJob, cfg config.Getter) prowapi.ProwJob {
+	var repos []string
+
+	if pj.Spec.Refs != nil {
+		repos = []string{pj.Spec.Refs.Org + "/" + pj.Spec.Refs.Repo}
+	}
+
+	// Attempt to lookup ProwJob from JobConfig by type, repo and name.
+	switch pj.Spec.Type {
+	case prowapi.PresubmitJob:
+		for _, pre := range cfg().AllStaticPresubmits(repos) {
+			if pre.Name == pj.Spec.Job {
```

Comment:
This also needs to take the branch into consideration (i.E. `pj.Spec.Refs.BaseRef`) because we unfortunately allow two jobs with the same name as long as they do not run on the same branch (Which we can not easily change, because it might cause a lot of breakages).

(And while you are at it, could you add to the godoc of `Refs.BaseRef` that its just a plain branch name? I look everytime at live prowjobs to find out if its  `refs/$branch` or just `$branch`)

Reply:
Sure thing, I will additionally include a branchname match and update the godoc.

Reply:
Oh, and add tests too 😉 .